app/controllers/achievements_controller.rb

```diff
@@ -6,7 +6,12 @@ def create
     @achievement = current_user.achievements.build(achievement_params)
 
     if @achievement.save && @achievement.transition_to(:drafted)
```

Comment:
Should we move the transition from this bit down into the if statement, that then removes my other comment about the need to go from drafted to enrolled?